engine/orchestration/src/com/cloud/vm/VirtualMachineManagerImpl.java

```diff
@@ -1890,6 +1890,13 @@ public void doInTransactionWithoutResult(final TransactionStatus status) throws
                     if (!stateTransitTo(vm, VirtualMachine.Event.DestroyRequested, vm.getHostId())) {
```

Comment:
Is this change only about the change in the state in the DB? Or is that `stateTransitTo` going to execute the commands to actually remove the volume?

Reply:
It is going to signal listeners through a publish-subscribe. I suppose one of them will somehow automagically call the hypervisor to destroy the VM. The new code below does the same for expunge, if expunge flag is set which is only so in one of the three places the method is called. The other two have a hardcoded false in place.

Reply:
Ok, thanks for the feedback. I still have some doubts though.

Why did you use the `else`?
I mean, instead of `ELSE/IF (expunge)`, you can simply use `if (expunge)`.

Reply:
because I don't want to do the expunge when the destroy fails or is refused.

Reply:
if it fails, or if it is refused (I have no idea what it means), an exception is thrown (I am assuming that for these cases the method will return false). Therefore, there is no need for the `else` block.

Reply:
the method returns a boolean and throws an exception when it returns false. I take action only on the true case. I guess it's a matter of style.

Reply:
but in the true case, we do not execute the IF body, and we proceed, which is the same as not using the ELSE block, and adding the IF(expunge) directly because the FALSE case will always throw an exception.

For it is not a matter of style, it is a matter of removing a nested condition IF/ELSE/IF, but sure we can leave it as is.

Reply:
relying on a mid method exit to occur is not my style. I'd like to see structure and I do not think it is readable like that, either. sorry.